src/System.Net.Requests/tests/AssemblyInfo.cs

```diff
@@ -0,0 +1,7 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+[assembly: CollectionBehavior(CollectionBehavior.CollectionPerAssembly, DisableTestParallelization = true, MaxParallelThreads = 1)]
```

Comment:
>Why do they hang rather than just queue up to run when the older connections complete? Or do the tests end up leaving a whole bunch of connections permanently blocked?

I think the tests hang because they are written using .GetAwaiter().GetResult(), i.e. they are blocking threads. If the tests were written "better" using more asynchrony then perhaps they would not hang. So, this is really a test bug.